src/Akka.Persistence.PostgreSql/Journal/PostgreSqlQueryExecutor.cs

```diff
@@ -105,7 +104,7 @@ protected override void WriteEvent(DbCommand command, IPersistentRepresentation
         private static string QualifiedName(IPersistentRepresentation e)
         {
             var type = e.Payload.GetType();
-            return type.FullName + ", " + type.Assembly.GetName().Name;
+            return type.AssemblyQualifiedName;
```

Comment:
Unfortunately this is not the same i.e:

- `type.FullName + ", " + type.Assembly.GetName().Name` (current dev) returns *"System.Int32, mscorlib"*
- `type.AssemblyQualifiedName`  returns *System.Int32, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089*

The problem is that, how the `Type.GetTyoe` will behave in face of upgrading the assembly? Example: we have stored an event of type T from assembly versioned as 1.0.0.0, later on we perform an upgrade - the type definition hasn't changed - to version 2.0.0.0 of that assembly. When we want to restore it, will `Type.GetType` work? Initial check on `int` type confirms it works, however I cannot say it will work in each case.

Note: this also must work in another direction (i.e. if user downgrades assembly version for some reason and wants to replay events from higher version).

Another matter is a length of the manifest - keeping version, culture and token for every single event will add around 67 extra nvarchars per event type. In some cases this may break existing code (as manifest column has limited length and people can have quite lenghty namespaces).

Reply:
Getting the assembly name in dotnet core using this code:
```
type.FullName + ", " + type.GetTypeInfo().Assembly.GetName().Name;
```
returns: _"System.String, System.Private.CoreLib"_

It also returns a full namespace. Is that ok?

Reply:
just use `type.TypeQualifiedName`

Reply:
That's even better :)